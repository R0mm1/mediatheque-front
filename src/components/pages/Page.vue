<template>
    <div id="med_page">

        <div id="med_page_header">
            <slot name="med_page_header">
                <Header :no-user="noUser" :custom-header-message="customHeaderMessage"></Header>
            </slot>
        </div>

        <div id="med_page_body">
            <slot name="med_page_body"></slot>
        </div>

        <div id="med_page_footer">
            <slot name="med_page_footer"></slot>
        </div>

    </div>
</template>

<script>
    export default {
        name: "Page",
        props: {
            noUser: {default: false},
            customHeaderMessage: {default: ''}
        },
        components: {
            Header: () => import('./Header')
        }
    }
</script>

<style scoped>
    #med_page {
        height: 100%;
        display: flex;
        flex-direction: column;
    }

    #med_page_body {
        overflow: auto;
        position: relative;
        flex: 1;
    }
</style>