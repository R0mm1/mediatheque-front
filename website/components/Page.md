# Page

## Props

<!-- @vuese:Page:props:start -->
|Name|Description|Type|Required|Default|
|---|---|---|---|---|
|noUser|-|—|`false`|false|
|customHeaderMessage|-|—|`false`|''|

<!-- @vuese:Page:props:end -->


## Slots

<!-- @vuese:Page:slots:start -->
|Name|Description|Default Slot Content|
|---|---|---|
|med_page_header|-|-|
|med_page_body|-|-|
|med_page_footer|-|-|

<!-- @vuese:Page:slots:end -->


