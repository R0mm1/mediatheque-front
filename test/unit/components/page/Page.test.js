import {mount} from '@vue/test-utils'
import Page from "@/components/pages/Page.vue";

describe('Page', () => {

    const factory = (withCustomHeader) => {
        if (typeof withCustomHeader === 'undefined') withCustomHeader = false;

        const slots = {
            med_page_body: '<div>Page Body</div>',
            med_page_footer: '<div>Page Footer</div>'
        };

        if (withCustomHeader) {
            slots.med_page_header = '<div>Page Header</div>';
        }

        return mount(Page, {
            slots: slots
        });
    };

    it('renders correctly with custom header', () => {
        expect(factory(true).element).toMatchSnapshot();
    });

    it('renders correctly with Header', () => {
        expect(factory(false).element).toMatchSnapshot();
    });
});
