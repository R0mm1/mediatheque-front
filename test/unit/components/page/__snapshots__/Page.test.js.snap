// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page renders correctly with Header 1`] = `
<div
  class=""
  id="med_page"
>
  <div
    id="med_page_header"
  >
    <header
      id="vueHeader"
    >
      <div
        id="mainLogo"
      >
        <img
          alt="Médiathèque"
          src="/images/logo.png"
        />
      </div>
       
      <!---->
       
      <div
        id="customHeaderMessage"
      >
        
      </div>
       
      <!---->
    </header>
  </div>
   
  <div
    id="med_page_body"
  >
    <div>
      Page Body
    </div>
  </div>
   
  <div
    id="med_page_footer"
  >
    <div>
      Page Footer
    </div>
  </div>
</div>
`;

exports[`Page renders correctly with custom header 1`] = `
<div
  class=""
  id="med_page"
>
  <div
    id="med_page_header"
  >
    <div>
      Page Header
    </div>
  </div>
   
  <div
    id="med_page_body"
  >
    <div>
      Page Body
    </div>
  </div>
   
  <div
    id="med_page_footer"
  >
    <div>
      Page Footer
    </div>
  </div>
</div>
`;
